politeia.go

```diff
@@ -0,0 +1,223 @@
+func (p *Politeia) GetVoteStatusRaw(censorshipToken string) (*VoteStatus, error) {
+	proposal, err := p.GetProposalRaw(censorshipToken)
+	if err != nil {
+		return nil, err
+	}
+
+	return &proposal.VoteStatus, nil
+}
+
+// GetVoteStatus returns the result of GetVoteStatusRaw as a JSON string
+func (p *Politeia) GetVoteStatus(censorshipToken string) (string, error) {
+	return processResult(p.GetVoteStatusRaw(censorshipToken))
+}
+
+func (p *Politeia) count(category ProposalCategory) (int, error) {
+	var proposals []Proposal
+
+	err := p.db.Select(q.Eq("Category", category)).Find(&proposals)
+	if err != nil {
+		return 0, fmt.Errorf("error fetching number of pre proposals: %s", err.Error())
+	}
+
+	return len(proposals), nil
+}
+
+// CountPreRaw gets the total count of proposals in discussion
+func (p *Politeia) CountPreRaw() (int, error) {
+	return p.count(PreProposals)
+}
+
+// CountPre returns the result of CountPreRaw as a JSON string
+func (p *Politeia) CountPre() (string, error) {
+	return processResult(p.CountPreRaw())
+}
+
+// CountActiveRaw gets the total count of active proposals
+func (p *Politeia) CountActiveRaw() (int, error) {
+	return p.count(ActiveProposals)
+}
+
+// CountActive returns the result of CountActiveRaw as a JSON string
+func (p *Politeia) CountActive() (string, error) {
+	return processResult(p.CountActiveRaw())
+}
+
+// CountApprovedRaw gets the total count of approved proposals
+func (p *Politeia) CountApprovedRaw() (int, error) {
+	return p.count(ApprovedProposals)
+}
+
+// CountApproved returns the result of CountApprovedRaw as a JSON string
+func (p *Politeia) CountApproved() (string, error) {
+	return processResult(p.CountApprovedRaw())
+}
+
+// CountRejectedRaw gets the total count of rejected proposals
+func (p *Politeia) CountRejectedRaw() (int, error) {
+	return p.count(RejectedProposals)
+}
+
+// CountRejected returns the result of CountRejected as a JSON string
+func (p *Politeia) CountRejected() (string, error) {
+	return processResult(p.CountRejectedRaw())
+}
+
+// CountAbandonedRaw gets the total count of abandoned proposals
+func (p *Politeia) CountAbandonedRaw() (int, error) {
+	return p.count(AbandonedProposals)
+}
+
+// CountAbandoned returns the result of CountAbandonedRaw as a JSON string
+func (p *Politeia) CountAbandoned() (string, error) {
+	return processResult(p.CountAbandonedRaw())
+}
+
+func (p *Politeia) prepareQuery(category ProposalCategory, offset, limit int32, newestFirst bool) (query storm.Query) {
+	switch category {
+	case AllProposals:
+		query = p.db.Select(
+			q.True(),
+		)
+	default:
+		query = p.db.Select(
+			q.Eq("Category", category),
+		)
+	}
+
+	if offset > 0 {
+		query = query.Skip(int(offset))
+	}
+
+	if limit > 0 {
+		query = query.Limit(int(limit))
+	}
+
+	if newestFirst {
+		query = query.OrderBy("Timestamp").Reverse()
+	} else {
+		query = query.OrderBy("Timestamp")
+	}
+
+	return
+}
+
+func processResult(result interface{}, err error) (string, error) {
```

Comment:
Make this function part of `Politeia` struct since it is specifically for proposal. it can also be renamed to `MarshalResult`.